### PR TITLE
Fix highlighting on single-character Rust consts

### LINKED ISSUE #3922: Syntax highlighting wrong for one character constants in Rust
### Summary

Constants with one character names are incorrectly highlighted in Rust 
They both have the same tree-sitter scopes: 
Definition: `scopes: ["source_file", "const_item", "identifier"] `
Usage:  `scopes: ["source_file", "function_item", "block", "let_declaration", "identifier"]`

image

```rust
const NUM: usize = 5;
const N: usize = 4;

fn f() {
    let a = N;
    let b = NUM;
}

```

### Reproduction Steps

 

I tried this:

1. `hx file.rs` file with code from above

I expected this to happen:
one letter constant "N" is highlighted the same as other constants

Instead, this happened:
It is highlighted the same as a "type_identifier"

### Helix log

Nothing

### Platform

Linux

### Terminal Emulator

Gnome console

### Helix Version

22.08.1

### PR DESCRIPTION
**Pull request by gavincrawford**
Resolves #3922 with the fix suggested by @the-mikedavis

### CODE REVIEW

File: runtime/queries/rust/highlights.scm [hunk 1]
```diff
@@ -183,7 +183,7 @@
 ; -------
 
 ((identifier) @constant
- (#match? @constant "^[A-Z][A-Z\\d_]+$"))
+ (#match? @constant "^[A-Z][A-Z\\d_]*$"))
```
Comment (kirawi) [inline runtime/queries/rust/highlights.scm:187]:
```suggestion
(const_item
  name: (identifier) @constant
)

```

**Comment by gavincrawford** [comment]
Screenshot from 2022-09-21 09-57-46
@kirawi After your changes, everything seems to be working correctly, thank you!
Take a look.

File: runtime/queries/rust/highlights.scm [hunk 2]
```diff
@@ -183,7 +183,10 @@
...
+(const_item
+  name: (identifier) @constant
+)
```
[the remark this answers is not in the document]
Comment (gavincrawford) [reply runtime/queries/rust/highlights.scm:189]: I see, good catch. I can make a commit later today to address this, if you don't get to it first :-)

[the remark this answers is not in the document]
Comment (gavincrawford) [reply runtime/queries/rust/highlights.scm:189]: @the-mikedavis Do you think we should remove that stanza entirely, due to the fact that all constants are expected to be declared in all caps by Rust's formatting rules?

[the remark this answers is not in the document]
Comment (the-mikedavis) [reply runtime/queries/rust/highlights.scm:189]: Yep I think we're ok to remove this stanza

**Review by the-mikedavis — APPROVED** [review]

**Comment by the-mikedavis** [comment]
Thanks!

**Pull request merged**